File: algorithms/selection-sort/selection_sort.js
// Input: array of integers
// Output: sorted array of integers
// Time Complexity: O(N^2)

const selectionSort = (arr) => {
  for (let i = 0; i < arr.length - 1; i++) {
    let lowestValueIndex = i;

    for (let j = i + 1; j < arr.length; j++) {
      if (arr[j] < arr[lowestValueIndex]) {
        lowestValueIndex = j;
      }
    }

    if (lowestValueIndex != i) {
      let temp = arr[lowestValueIndex];
      arr[lowestValueIndex] = arr[i];
      arr[i] = temp;
    }
  }

  return arr;
};

const nums = [13, 2, 8, 7, 10, 1, 17, 15, 18, 20, 4, 3];
console.log(selectionSort(nums));

File: algorithms/selection-sort/selection_sort.py
# Input: array of integers
# Output: sorted array of integers
# Time Complexity: O(N^2)

def selection_sort(arr):
  for i in range(len(arr)-1):
    lowest_value_index = i 

    for j in range(i+1, len(arr)):
      if arr[j] < arr[lowest_value_index]:
        lowest_value_index = j 
    
    if lowest_value_index != i:
      arr[i], arr[lowest_value_index] = arr[lowest_value_index], arr[i] 
  
  return arr
      
nums = [13, 2, 8, 7, 10, 1, 17, 15, 18, 20, 4, 3]
print(selection_sort(nums))